diff -Naur go-ethereum-1.10.6/ethclient/ethclient.go go-ethereum-1.10.6-with-receipts/ethclient/ethclient.go
--- go-ethereum-1.10.6/ethclient/ethclient.go	2021-07-22 09:44:28.000000000 -0500
+++ go-ethereum-1.10.6-with-receipts/ethclient/ethclient.go	2021-08-24 09:34:50.475745620 -0500
@@ -284,6 +284,17 @@
 	return r, err
 }
 
+func (ec *Client) GetBlockReceipts(ctx context.Context, txHash common.Hash) (types.Receipts, error) {
+	var r types.Receipts
+	err := ec.c.CallContext(ctx, &r, "eth_getBlockReceipts", txHash)
+	if err == nil {
+		if r == nil {
+			return nil, ethereum.NotFound
+		}
+	}
+	return r, err
+}
+
 type rpcProgress struct {
 	StartingBlock hexutil.Uint64
 	CurrentBlock  hexutil.Uint64
diff -Naur go-ethereum-1.10.6/internal/ethapi/api.go go-ethereum-1.10.6-with-receipts/internal/ethapi/api.go
--- go-ethereum-1.10.6/internal/ethapi/api.go	2021-07-22 09:44:28.000000000 -0500
+++ go-ethereum-1.10.6-with-receipts/internal/ethapi/api.go	2021-08-24 09:34:40.071662525 -0500
@@ -1646,6 +1646,14 @@
 	return fields, nil
 }
 
+func (s *PublicTransactionPoolAPI) GetBlockReceipts(ctx context.Context, hash common.Hash) (types.Receipts, error) {
+	receipts, err := s.b.GetReceipts(ctx, hash )
+	if err != nil {
+		return nil, err
+	}
+	return receipts,nil
+}
+
 // sign is a helper function that signs a transaction with the private key of the given address.
 func (s *PublicTransactionPoolAPI) sign(addr common.Address, tx *types.Transaction) (*types.Transaction, error) {
 	// Look up the wallet containing the requested signer
